### ISSUE TITLE
Class names for spans similar to divs

### ISSUE THREAD

**Issue by dz4k** [jgm/djot#239]
Many Markdown implementations support this syntax:

```md
A :dfn[hypermedia control] is an element in a hypermedia that describes (or controls) some sort of interaction, often with a remote server, by encoding information about that interaction directly and completely within itself.
```

which would presumably be transformed by a filter into an HTML `<dfn>` element.

The current syntax for such a thing would be

```md
a [hypermedia control]{.dfn}
```
which is more verbose.

I think this should be supported for links as well:

```md
Let us consider these two defining hypermedia elements (that is the two defining :ref[hypermedia controls][hypermedia control]) of HTML, the anchor tag and the form tag, in a bit of detail.
```

**Comment by Omikhleia** [comment]
> Many Markdown implementations support this syntax (...)

References?

**Comment by dz4k** [comment]
@Omikhleia 

- Here's a markdown-it plugin: https://github.com/hilookas/markdown-it-directive
- remark impl: https://github.com/remarkjs/remark-directive
- micromark (i never understood if this was separate from remark or not) https://github.com/micromark/micromark-extension-directive
- Discussion on CommonMark forum: https://talk.commonmark.org/t/generic-directives-plugins-syntax/444

**Comment by jgm** [comment]
You'd save two characters of typing, at the cost of a new construct, less uniformity, and more complex parsing. Not sure it's worth it.

**Comment by matklad** [comment]
Yeah, I [use `[]{.dfn}`](https://github.com/matklad/matklad.github.io/blob/63e7278254c4a8386ea383bba0e0417d3c837a73/djot.ts#L183-L186) syntax for my blog, and it works quite OK. 

>You'd save two characters of typing

I would say that another aspect here is the position of `dfn`: if "tag name" comes first, its a bit more natural for this use-case. The distinction between "tag name" and "attribute" is very fuzzy of course, but I do feel certain friction in practice here. Leading tag would also be a bit more autocompletion / snippet friendly. 

I actually came to this comment here to concur with "not needed", but now I am thinking, wouldn't it be cool if both of these:

```
inline :kbd[ctrl+c]

::: details
block
:::
```

produced a `<kbd></kbd>` / `<details></details>` in the HTML output without any custom readers/writers? See also https://github.com/jgm/djot/issues/146.

**Comment by Omikhleia** [comment]
@matklad Regarding your new idea above, I am not sure what it would then mean for non-HTML renderers...
Also, the implication now is the `:em[doe]` would give `<em>doe</em>`  -- isn't it just reinventing a syntax for HTML? :cat2:

**Comment by matklad** [comment]
Fleshed out the idea in significantly more detail in

https://github.com/jgm/djot/issues/240

>Also, the implication now is the :em[doe] would give <em>doe</em> -- isn't it just reinventing a syntax for HTML? 

Yup, that's the point. The XML model with tag-name,attributes,children is great for markup, and a lot of formats support that (DocBook XML, SILE XML, LaTeX environments). If the target format doesn't have a direct support for tag names, it could be rendered exactly as it is today, as a class.

**Comment by Omikhleia** [comment]
> The XML model with tag-name,attributes,children is great for markup

But what this does it have to do with Djot markup then? 
Honestly I am lost now :)
